### ISSUE TITLE
Some minor errors in repo / flake8

### ISSUE THREAD

**Issue by max-sixty** [pydata/xarray#2990]
Currently we use pycodestyle: https://github.com/pydata/xarray/blob/ccd0b047ea8ca89c68ab6cfa942557e676e7d402/.travis.yml#L63

I think we used to use flake8. I can't find / remember the reason we moved to pycodestyle. 

master has some non-trivial issues that flake would catch, including a test overwritting another and undefined variables:
```
flake8 xarray --ignore=I,W503,W504,F401,E265,E402

xarray/core/options.py:62:8: F632 use ==/!= to compare str, bytes, and int literals
xarray/core/dataset.py:4148:69: F821 undefined name 'dim'
xarray/backends/netCDF4_.py:177:12: F632 use ==/!= to compare str, bytes, and int literals
xarray/tests/test_dataarray.py:1264:9: F841 local variable 'foo' is assigned to but never used
xarray/tests/test_dataarray.py:1270:18: F821 undefined name 'x'
xarray/tests/test_dataarray.py:1301:5: F811 redefinition of unused 'test_reindex_fill_value' from line 1262
xarray/tests/test_dataarray.py:1647:16: F632 use ==/!= to compare str, bytes, and int literals
xarray/tests/test_dataarray.py:1648:16: F632 use ==/!= to compare str, bytes, and int literals
xarray/tests/test_dataset.py:4759:8: F632 use ==/!= to compare str, bytes, and int literals
xarray/tests/test_dataset.py:4761:10: F632 use ==/!= to compare str, bytes, and int literals
xarray/tests/test_distributed.py:62:9: F811 redefinition of unused 'loop' from line 12
xarray/tests/test_distributed.py:92:9: F811 redefinition of unused 'loop' from line 12
xarray/tests/test_distributed.py:117:49: F811 redefinition of unused 'loop' from line 12
xarray/tests/test_distributed.py:141:53: F811 redefinition of unused 'loop' from line 12
xarray/tests/test_distributed.py:152:51: F811 redefinition of unused 'loop' from line 12
```

Happy to fix these in a PR. For ensuring these don't crop up again, any objection to flake8?

**Comment by shoyer** [comment]
I would be very happy to switch back to flake8, but we should also switch `pep8speaks` to use flake8, too, which it now supports!

**Comment by max-sixty** [comment]
Closed by #3010
